### DISCUSSION TITLE
Getting started

### DISCUSSION THREAD

**Discussion by BenjyNStrauss** [sailuh/kaiaulu#305] [Q&A] [answered]
I'm trying to get started with Kaiaulu, but I'm not sure I'm correctly understanding what is going on.
(1) What do I do with the "Project Configuration File"?  Should I copy it and save it?  And if so, do I need to call it anything specific?
(2) Do I download the git repo that I want to search through?  Or does the Kaiaulu search the web?

Thanks,
Benjamin Strauss

**Comment by carlosparadis** [comment, marked as answer]
(1) Generally, software engineering research requires you to go to an open-source website and collect a number of information. @rnkazman will sooner or later ask you to, for example, collect bug data. You will also need the git log, and so on. Rather than hardcoding all this information, which is unique to a project, into code, making it nonreusable, we annotate it in a file. That is the project configuration file. Think of it as a summary of all the information the tool needs to run the data on your machine.

(2) Kaiaulu is an offline tool, although some functionality may download data from APIs. The Notebook I pointed you to explain a bit of the configuration file organization of a project: http://itm0.shidler.hawaii.edu/kaiaulu/articles/gitlog_showcase.html 

You should only need to copy and paste if you are using an existing project. Otherwise, you need to create your own and download and place the files accordingly. You have to download the files to your computer and specify where on the config file you placed them. The notebook is not doing much: It just reads the content of the configuration file, gets the information, and the functions along the way make a few syscalls to ingest the data into a table.
